src/pages/patch/patchTabs/Tasks.tsx

```diff
@@ -71,6 +74,82 @@ export const Tasks: React.FC<Props> = ({ taskCount }) => {
   }
   useNetworkStatus(startPolling, stopPolling);
 
+  const columnsTemplate: Array<ColumnProps<TaskResult>> = [
+    {
+      title: "Name",
+      dataIndex: "displayName",
+      key: TaskSortCategory.Name,
+      sorter: {
+        multiple: 4,
+      },
+      width: "40%",
+      className: "cy-task-table-col-NAME",
+      render: (name: string, { id }: TaskResult): JSX.Element => (
+        <TaskLink taskName={name} taskId={id} />
+      ),
+    },
+    {
+      title: "Patch Status",
+      dataIndex: "status",
+      key: TaskSortCategory.Status,
+      sorter: {
+        multiple: 4,
+      },
+      className: "cy-task-table-col-STATUS",
+      render: renderStatusBadge,
+    },
+    {
+      title: "Base Status",
+      dataIndex: "baseStatus",
+      key: TaskSortCategory.BaseStatus,
+      sorter: {
+        multiple: 4,
+      },
+      className: "cy-task-table-col-BASE_STATUS",
+      render: renderStatusBadge,
+    },
+    {
+      title: "Variant",
+      dataIndex: "buildVariant",
+      key: TaskSortCategory.Variant,
+      sorter: {
+        multiple: 4,
+      },
+      className: "cy-task-table-col-VARIANT",
+    },
+  ];
+  if (sorts.length === 0) {
+    // works around a quirk/bug in the antd table where default sorts don't
+    // get updated on subsequent renders
+    columnsTemplate[1].defaultSortOrder = "ascend";
+    columnsTemplate[2].defaultSortOrder = "ascend";
+  } else {
+    sorts.forEach((sort) => {
+      const direction =
+        sort.Direction === SortDirection.Desc ? "descend" : "ascend";
+      switch (sort.Key) {
+        case TaskSortCategory.Name: {
+          columnsTemplate[0].defaultSortOrder = direction;
+          break;
+        }
+        case TaskSortCategory.Status: {
+          columnsTemplate[1].defaultSortOrder = direction;
+          break;
+        }
+        case TaskSortCategory.BaseStatus: {
+          columnsTemplate[2].defaultSortOrder = direction;
+          break;
+        }
+        case TaskSortCategory.Variant: {
+          columnsTemplate[3].defaultSortOrder = direction;
+          break;
+        }
+        default:
```

Comment:
I think this logic could be broken out into its own util function. The reason your probably experience some weird errors with the defaultSorts not being reflected in the table is likely a result of 2 things.

Reply:
I actually did this previously, but switched to what you see now because it didn't work. I tried again with your suggestions (https://gist.github.com/john-m-liu/3cffc77e817b4f0eb80462c185558381) and see the same problem